### PR TITLE
removed bug, vulnerability and some code smells

### PR DESCRIPTION
**Pull request by paolovic**

### CODE REVIEW

File: src/main/java/ch/uzh/ifi/hase/soprafs22/game/tiles/TileBuilder.java [hunk 1]
```diff
@@ -71,13 +77,14 @@ public void setTileTraversingCost(int traversingCost) {
 
     //TODO create a UnitBuilder analogous to the TileBuilder
     @Override
-    public void setTileUnit(LinkedHashMap<String, Object> unitStream) {
+    public void setTileUnit(Map<String, Object> unitStream) {
         String unitType = (String) unitStream.get("type");
         Unit unit = new Unit();
         switch (unitType) {
             case "archer" -> unit.setType(UnitType.ARCHER);
```
Comment (apple00juice) [inline src/main/java/ch/uzh/ifi/hase/soprafs22/game/tiles/TileBuilder.java:84]: I don't know switches with -> but don't wee need "break;" else it triggers always default?

**Review by apple00juice — APPROVED** [review]
Looks good 👍

File: src/main/java/ch/uzh/ifi/hase/soprafs22/game/tiles/TileBuilder.java (hunk 1 above)
Comment (paolovic) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/game/tiles/TileBuilder.java:84]: nope, new fancy java syntax
(which I didn't know, IntelliJ let me shine^^)

Comment (apple00juice) [reply src/main/java/ch/uzh/ifi/hase/soprafs22/game/tiles/TileBuilder.java:84]: uhh nice, indeed very fancy 🎩 🧐

**Pull request merged**